pandas/tseries/period.py

```diff
@@ -281,6 +281,16 @@ def __contains__(self, key):
             return False
         return key.ordinal in self._engine
 
+    def __array_wrap__(self, result, context=None):
```

Comment:
Raising ``TypeError`` in ``__array_wrap__`` affects to arithmetic using ``np.array``. Currently, it works:

```
pd.PeriodIndex(['2011', '2012'], freq='A') + np.array([1, 2])
# PeriodIndex(['2012', '2014'], dtype='int64', freq='A-DEC')
```

So it can't be prohibited?